### ISSUE TITLE
ts-jsx for browserify

### ISSUE THREAD

**Issue by davidreher** [jbrantly/ts-jsx-loader#5]
Hey,
 
I saw James talk at react conf about using typescript and jsx. I really liked it and currently we use typescript for the parts of our application which are not react components. Since we are using browserify I wanted to implement a plugin similar to ts-jsx-loader. After I implemented it, the typescript compiler always complains about the imported views. The views are defined as:
```
export class X extends React.Component<any, any, any> {​ ... }
```
and then imported as
```
import X = require('./x');

React.render(
  React.jsx(`<X />`),
  document.body
);
```

The typescript compiler then complains about this:
 
Type 'typeof "src/x"' is not assignable to type 'new (props: any, context: any) => Component<any, any, any>'.

Can you give me any hint, how to get around that issue? I tried using the defintions from your talk, as well as the definitions in definetly typed ... nothing works.

**Comment by jbrantly** [comment]
If I'm not mistaken the `export class X` construct exports as a property of the module. So usage would be something like

```
import Xmodule = require('./x');

React.render(
  React.jsx(`<Xmodule.X />`),
  document.body
);
```
If you want to export the class *as* the module, use `export =`. So

```
class X extends React.Component<any, any, any> {​ ... }
export = X;
```

**Comment by davidreher** [comment]
:( thanks for the help, that really did it. I did not know that it would mean such a different thing ...

**Comment by davidreher** [comment]
By the way: do you know how to work with the typings provided by DefinitelyTyped?

**Comment by jbrantly** [comment]
I need to update for the new v13 typings on DefinitelyTyped. Things changed a bit and my React.jsx typings don't work.

**Comment by jbrantly** [comment]
I just published v0.2.0 which has updated definitions for React v0.13
